graphql_compiler/fast_introspection.py

```diff
@@ -0,0 +1,300 @@
+graphql_sync must parse and traverse the query document to identify the next field to resolve,
+whereas in this module, everything is hardcoded. This may be the source of the severe
+performance difference.
+
+This module is fully dependent on the introspection query returned by `get_introspection_query`
+from graphql-core (with default parameters). Changes to that query will need to be
+reflected in this module.
+"""
+
+from typing import Any, Dict, Optional, Tuple, cast
+
+from graphql import (
+    ExecutionResult,
+    GraphQLAbstractType,
+    GraphQLArgument,
+    GraphQLDirective,
+    GraphQLEnumType,
+    GraphQLEnumValue,
+    GraphQLField,
+    GraphQLNamedType,
+    GraphQLObjectType,
+    GraphQLSchema,
+    GraphQLType,
+    introspection_types,
+    is_abstract_type,
+    validate_schema,
+)
+
+
+_introspection_query = """
+query IntrospectionQuery {
+    __schema {
+        queryType { name }
+        mutationType { name }
+        subscriptionType { name }
+        types {
+            ...FullType
+        }
+        directives {
+            name
+            description
+            locations
+            args {
+                ...InputValue
+            }
+        }
+    }
+}
+
+fragment FullType on __Type {
+    kind
+    name
+    description
+    fields(includeDeprecated: true) {
+        name
+        description
+        args {
+            ...InputValue
+        }
+        type {
+            ...TypeRef
+        }
+        isDeprecated
+        deprecationReason
+    }
+    inputFields {
+        ...InputValue
+    }
+    interfaces {
+        ...TypeRef
+    }
+    enumValues(includeDeprecated: true) {
+        name
+        description
+        isDeprecated
+        deprecationReason
+    }
+    possibleTypes {
+        ...TypeRef
+    }
+}
+
+fragment InputValue on __InputValue {
+    name
+    description
+    type { ...TypeRef }
+    defaultValue
+}
+
+fragment TypeRef on __Type {
+    kind
+    name
+    ofType {
+        kind
+        name
+        ofType {
+            kind
+            name
+            ofType {
+                kind
+                name
+                ofType {
+                    kind
+                    name
+                    ofType {
+                        kind
+                        name
+                        ofType {
+                            kind
+                            name
+                            ofType {
+                                kind
+                                name
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+}
+"""
+
+
+def remove_whitespace_from_query(query: str) -> str:
+    """Return an equivalent query with spaces and newline characters removed."""
+    return query.replace(" ", "").replace("\n", "")
+
+
+whitespace_free_introspection_query = remove_whitespace_from_query(_introspection_query)
```

Comment:
nit: Thoughts on making these use underscore-prefixed names? I know tests need to import them and it's a bit iffy to import underscored names, but it makes it very explicit that these parts are not part of the public API and shouldn't be used from the outside (even from other places in the compiler other than this module's tests).

Reply:
Good suggestion, I think it's good to be explicit